File: src/main/java/no/nav/veilarbvedtaksstotte/kafka/dto/KafkaVedtakSendt.java
package no.nav.veilarbvedtaksstotte.kafka.dto;

import lombok.Data;
import lombok.experimental.Accessors;
import no.nav.veilarbvedtaksstotte.domain.vedtak.Hovedmal;
import no.nav.veilarbvedtaksstotte.domain.vedtak.Innsatsgruppe;

import java.time.LocalDateTime;

@Data
@Accessors(chain = true)
public class KafkaVedtakSendt {
    long id;
    LocalDateTime vedtakSendt;
    Innsatsgruppe innsatsgruppe;
    Hovedmal hovedmal;
    String aktorId;
    String enhetId;
}

File: src/main/java/no/nav/veilarbvedtaksstotte/domain/kafka/KafkaVedtakSendt.kt
package no.nav.veilarbvedtaksstotte.domain.kafka

import no.nav.veilarbvedtaksstotte.domain.vedtak.Hovedmal
import no.nav.veilarbvedtaksstotte.domain.vedtak.Innsatsgruppe
import no.nav.veilarbvedtaksstotte.domain.vedtak.Vedtak
import java.time.LocalDateTime

data class KafkaVedtakSendt(
    val id: Long,
    val vedtakSendt: LocalDateTime,
    val innsatsgruppe: Innsatsgruppe,
    val hovedmal: Hovedmal,
    val aktorId: String,
    val enhetId: String
)

fun Vedtak.toKafkaVedtakSendt(): KafkaVedtakSendt {
    return KafkaVedtakSendt(
        id = id,
        vedtakSendt = vedtakFattet,
        innsatsgruppe = innsatsgruppe,
        hovedmal = hovedmal,
        aktorId = aktorId,
        enhetId = oppfolgingsenhetId
    )
}